tools/build_and_install.py

```
@@ -27,6 +28,49 @@ def t(key: str, **kwargs) -> str:
     return _local_t(key, **kwargs)
 
 
+def _timing(label: str, start: float) -> None:
+    print(f"  [timing] {label}: {time.monotonic() - start:.1f}s", flush=True)
+
+
+def _find_ccache(root_dir: Path) -> str | None:
+    """找可用的 ccache：优先工作区官方版 (.cache/ccache-bin)，退回 PATH；需 >= 4.6 (MSVC 支持)。"""
```

Ruff (0.16.2)

[warning] Docstring contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF002)